### PR TITLE
feat: INTEG-554 add user consent types to UserAPI

### PR DESCRIPTION
**Pull request by mgoudy91**
# Purpose of PR

This change supports [a necessary change](https://github.com/contentful/experience-packages/pull/2125) to the UserAPI to add user consent information.

More context for this can be found in "Proposal for experience-sdk changes to provide tracking to Google Analytics 4 app"

## PR Checklist

- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Typescript typings are added/updated/not required

### CODE REVIEW

File: lib/types/api.types.ts [hunk 1]
```diff
@@ -353,4 +355,5 @@ export interface ConnectMessage {
   }
   fieldInfo: EntryFieldInfo[]
   field?: FieldInfo
+  trackingClient: any
```
Comment (mgoudy91) [inline lib/types/api.types.ts:358]: rm this

**Review by jsdalton — APPROVED** [review]
💯

**Review by colomolo — APPROVED** [review]

File: lib/types/api.types.ts [hunk 2]
```diff
@@ -22,6 +22,26 @@ import { EntryFieldInfo, FieldInfo } from './field.types'
...
+  | 'MARKETING'
+  | 'PERSONALIZATION'
+  | 'OPT_OUT'
+  | 'STORAGE'
+
+declare type ConsentData = {
+  uuid: string
+  expirationDate: string | null
+  consentRecord: Record<ConsentRecordKey, 'ACCEPT' | 'DENY'>
+  rawConsentRecord: string
```
Comment (andipaetzold) [inline lib/types/api.types.ts:39]: Why expose `uuid` and `rawConsentRecord`? I think that will cause confusion for the users.

File: lib/types/api.types.ts [hunk 3]
```diff
@@ -40,6 +60,7 @@ export interface UserAPI {
     admin: SpaceMembership['admin']
     roles: Pick<Role, 'name' | 'description'>[]
   }
+  consentRecord?: UserConsent
```
Comment (andipaetzold) [inline lib/types/api.types.ts:63]: Can we add a jsdoc explaining what the `consentRecord` is? Also, why optional?

**Pull request closed without merging**